src/main/java/me/chrr/scribble/book/RichSelectionManager.java

```diff
@@ -174,53 +183,41 @@ private void applyFormatting(
             RichText text = this.textGetter.get()
                     .applyFormatting(start, end, newColor, addModifiers, removeModifiers);
             this.textSetter.accept(text);
-        } else {
-            if (newColor != null) {
-                this.color = newColor;
-            }
-
-            this.modifiers.addAll(addModifiers);
-            this.modifiers.removeAll(removeModifiers);
         }
     }
 
-    public void updateSelectionFormatting() {
+    private void notifyCursorFormattingChanged() {
         if (this.textGetter == null) {
             // We're too early, abort.
+            // Can happen when the methods is called from the supper constructor
             return;
         }
 
         int start = Math.min(this.selectionStart, this.selectionEnd);
         int end = Math.max(this.selectionStart, this.selectionEnd);
-        Pair<Formatting, Set<Formatting>> format = this.textGetter.get().getCommonFormat(start, end);
-
-        this.color = format.getLeft();
-        this.modifiers = new HashSet<>(format.getRight());
+        Pair<@Nullable Formatting, Set<Formatting>> format = this.textGetter.get().getCommonFormat(start, end);
```

Comment:
I'm not sure if its from this PR, but something is wrong. Selecting a paragraph with multiple colors now gives the final color black, instead of `null`. I'm guessing somewhere down the line that `null` gets converted back to the default color black, while it should stay `null`, as no common color is selected when multiple colors are.